Fix compatibility with Julia v1.0

vyu opened the pull request:
In Julia v1.0, the use of `eachrow` and `isnothing` breaks the printing of benchmark results as they were only added in v1.1. This PR replaces them with compatible alternatives.

vchuravy commented:
Thanks for your PR! Could you add a test?

tecosaur commented on src/trials.jl:
```diff
@@ -455,7 +455,8 @@ function Base.show(io::IO, ::MIME"text/plain", t::Trial)
     print(io, "\n")
-    for histrow in eachrow(hist)
+    for r in axes(hist, 1)
+        histrow = view(hist, r, :)
         print(io, "\n", pad, "  ")
         for (i, bar) in enumerate(histrow)
```
Why not replace `enumerate(histrow)` with `enumerate(hist[r,:])` ?

vyu replied:
Sure, I've changed it to skip the `histrow` assignment. (`enumerate(view(hist, r, :))` to avoid the allocation with array slicing.)

tecosaur replied:
Cool.

vyu commented:
Yes, sure, I've added a test on the exact output of `show(::IO, ::MIME"text/plain", t::Trial)` for a simple `t`.

The test found a bug in the [previous bugfix](https://github.com/JuliaCI/BenchmarkTools.jl/commit/ed0096a29825d02682d0f617ab934531528e5824) to `remtrailingzeros` in that it no longer removed trailing zeros (the anchor needs to be a space since `timestr` is of the form `\d+\.\d{3} \w+`), so I've fixed that as well.

tecosaur commented on test/TrialsTests.jl:
```diff
@@ -233,4 +233,17 @@ else
 
 end
 
+trial = BenchmarkTools.Trial(BenchmarkTools.Parameters(), [1.0, 1.01], [0.0, 0.0], 0, 0)
```
Should this be its own commit?

vyu replied:
I'm not sure what you mean. This test is diagnostic against the current and the previous `remtrailingzeros` bugfixes by checking that "1.01 ns" is not printed as "1.010 ns" or "11 ns". It is also diagnostic against the bugfix for compability with Julia 1.0 by calling `show` on a `Trial`. It seems better to commit them together.

tecosaur replied:
Ah right, I just thought this was a random test.

tecosaur commented on src/trials.jl:
```diff
@@ -465,13 +466,7 @@ function Base.show(io::IO, ::MIME"text/plain", t::Trial)
-            replace(timestr, r"(\.\d+?)0+$" => s"\1")
-        end
-    end
+    remtrailingzeros(timestr) = replace(timestr, r"\.?0+ " => " ")
```
What's with the space in this regexp?

vyu replied:
Oh, you might have looked at this before I posted my comment that explains why I changed this function. To recap: `timestr` includes units (e.g., `1.010 ns`), so we need to anchor to a space instead of the end of line. In any case, the spaces on each side act as a positive lookahead, so this is equivalent to `r"\.?0+(*pla: )" => ""`. Without them the function would revert to unconditionally removing the decimal point and right-adjacent zeros (which might not be trailing).

tecosaur replied:
Ah right, I just thought this was a random test.

tecosaur replied:
I did look at this before reading your comment, this seems sensible.

vchuravy commented on test/TrialsTests.jl:
```diff
@@ -233,4 +233,17 @@ else
+ Time  (median):     1.005 ns             ┊ GC (median):    0.00%
+ Time  (mean ± σ):   1.005 ns ± 0.007 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%
+
+  █                                                       █  
```
@tecosaur one of the bars here is misaligned?

tecosaur replied:
It looks fine to me.

![image](https://user-images.githubusercontent.com/20903656/129333441-4d3a2c41-12d6-4bf1-b6a0-8a1a219efb38.png)
